packages/core/src/dialects/abstract/connection-manager.ts

```diff
@@ -175,8 +175,13 @@ export class AbstractConnectionManager<TConnection extends Connection = Connecti
     await this._initDatabaseVersion();
 
     try {
+
+      await this.sequelize.hooks.runAsync('beforePoolAcquire', { ...this.config, password: this.config.password || '' });
```

Comment:
I don't think `this.config` is the right value to provide to the hook. That config is global and can always be read through `sequelize.config`. I think it would make more sense to provide `options` instead

Reply:
So, we are talking about the `sequelize.options`,  that would be added instead of a `this.config`. Am I right?

Reply:
No I meant the options that this method receives, which will control things like from which pool the connection must come from (read replica or write pool)